MatrixSDK/Threads/MXThreadingService.swift

```diff
@@ -144,33 +144,58 @@ public class MXThreadingService: NSObject {
             return nil
         }
 
-        let filter = MXRoomEventFilter()
-        filter.relationTypes = [MXEventRelationTypeThread]
-        if onlyParticipated {
-            filter.relationSenders = [session.myUserId]
-        }
-        return session.matrixRestClient.messages(forRoom: roomId,
-                                                 from: "",
-                                                 direction: .backwards,
-                                                 limit: nil,
-                                                 filter: filter) { response in
-            switch response {
-            case .success(let paginationResponse):
-                guard let rootEvents = paginationResponse.chunk else {
-                    completion(.success([]))
-                    return
-                }
+        if session.homeserverCapabilities?.threads?.isEnabled == true {
+            //  homeserver supports threads
+            let filter = MXRoomEventFilter()
+            filter.relationTypes = [MXEventRelationTypeThread]
+            if onlyParticipated {
+                filter.relationSenders = [session.myUserId]
+            }
+            return session.matrixRestClient.messages(forRoom: roomId,
+                                                     from: "",
+                                                     direction: .backwards,
+                                                     limit: nil,
+                                                     filter: filter) { response in
+                switch response {
+                case .success(let paginationResponse):
+                    guard let rootEvents = paginationResponse.chunk else {
+                        completion(.success([]))
```

Comment:
Do these completions need to be dispatched on the main queue like the ones on the `else` branch below?

Reply:
Nope, they are already on main thread, as they are response from the network.